### ISSUE TITLE
Decoding h264 byte stream from websocket

### ISSUE THREAD

**Issue by vsranjitroshan** [ened/flutter_plugin_h264#5]
Is it possible to feed and decode a byte stream from a server? I saw in the example that after loading the mp4 samples, the byte buffer is being fed into the h264 decoder. Can we do the same with the stream from a websocket? Is decoding of inter-frame compressed stream possible?

**Comment by ened** [comment]
This should technically be possible.

We can introduce the concept of a H264 decoding session and feed the bytes to it. 

Error handling may be tricky. 

Also, this will only cover the video output currently, at any speed and without considerations for buffering (like a real video player). Is this what you are looking for?

**Comment by vsranjitroshan** [comment]
Yes, this is what I am looking for. I have implmented myself a decoder code that works. I am stuck at the part on how to get the actual image buffer from the getOutputBuffer call of the android mediaCodec. I have to send the image to flutter and render it there on screen. Can you please guide on how to do that?

**Comment by ened** [comment]
Split the problem into sub problems, like transferring the byte stream from Dart to the native side efficiently (maybe this needs `dart:ffi`), then we can start solving the next problem.

I think this plugin should expose a native widget and draw directly into it for this use case.
